universal-application-tool-0.0.1/test/FunctionalTest.java

```diff
@@ -31,7 +30,20 @@
 public class FunctionalTest extends WithApplication {
 
   protected Application provideApplication() {
-    return fakeApplication(inMemoryDatabase("default", ImmutableMap.of("MODE", "PostgreSQL")));
+    return fakeApplication(
+        ImmutableMap.of(
+            "db.default.driver",
+            "org.testcontainers.jdbc.ContainerDatabaseDriver",
+            "db.default.url",
+            /* This is a magic string.  The components of it are
+             * jdbc: the standard java database connection uri scheme
+             * tc: Testcontainers - the tool that starts a new container per test.
+             * postgresql: which container to start
+             * 9.6.8: which version of postgres to start
+             * ///: hostless URI scheme - anything here would be ignored
+             * databasename: the name of the db to connect to - any string is okay.
```

Comment:
Question: any idea why the database name is ignored? I'd expect it'd need to always be the name of the test database specified elsewhere.

Reply:
`tc` automatically creates the database so both the source and reference are defined by the same string.